### PR TITLE
Add retrain code with all the dependencies

### PR DESCRIPTION
**Pull request by me-diru**
- Created retrain_dataset_generation notebook for test, eval, predict dataset to retrain the model.
- Created retrain_model function with a static learning rate.
- Updated preprocessing.py to preprocess data outside the dataset_generation notebook.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,38 +1,185 @@
...
+pycountry==19.8.18
+pycparser==2.19
+pycryptodome==3.9.4
+Pygments==2.6.1
+PyGObject==3.34.0
+PyJWT==1.7.1
+pylint==2.4.4
+pyOpenSSL==19.1.0
+PyPDF2==1.26.0
+PyQt5-sip==4.19.20
```
Comment (yvanzo) [inline requirements.txt]: It’s very surprising, `4.19.20` is not even listed from <https://pypi.org/project/PyQt5-sip/#history>.

Please remove unneeded dependencies.

**Review by yvanzo — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 1 above)
Comment (yvanzo) [reply requirements.txt]: Thanks

**Review by yvanzo — APPROVED** [review]
Thanks for all the changes, that finally works!

For the record: Documentation is due with next pull request.
